### PR TITLE
Address mamba v2 incompatibilities

### LINKED ISSUE #366: Mamba example 6 seems to be failing
https://github.com/conda-incubator/setup-miniconda/actions/workflows/example-6.yml

### PR DESCRIPTION
**Pull request by jaimergp**
Closes https://github.com/conda-incubator/setup-miniconda/issues/366

Mamba v2 compatibility needs:

- Setting `MAMBA_ROOT_PREFIX` to the `base` env location.
- Using `mamba env create` instead of `mamba env update` if the target env doesn't exist. conda does create the target env with `env update` if it didn't exist 🤷 
- On Unix, symlink `bin/mamba` to `condabin` so it can be found in PATH later (same as https://github.com/conda-forge/mamba-feedstock/pull/251).
- On Windows, write the `condabin/mamba.bat` file from mamba 1.x, adjusting the path to `mamba.exe` as necessary.

### CODE REVIEW

**Comment by jaimergp** [comment]
Ok, this seems to work mostly in Unix, but mamba v2 gets "stuck" for Windows in this part:

```
...
  Linking requests-2.32.3-pyhd8ed1ab_0
  Linking jsonschema-4.23.0-pyhd8ed1ab_0
  Linking requests-toolbelt-1.0.0-pyhd8ed1ab_0
  Linking nbformat-5.10.4-pyhd8ed1ab_0
  Linking anaconda-client-1.12.3-pyhd8ed1ab_1
  Warning: warning  libmamba draining failed An existing connection was forcibly closed by the remote host.
  
  warning  libmamba draining failed An existing connection was forcibly closed by the remote host.
```

**Comment by jaimergp** [comment]
Disabling PYC compilation unlocks the hang, but then it hangs again during the `pip` installs. Seems that in both cases a hung subprocess on Windows is involved. I guess these code blocks:

- https://github.com/mamba-org/mamba/blob/31b23473e059b47c03b065e5592ea5a7cf77fc9b/libmamba/src/api/pip_utils.cpp#L173
- https://github.com/mamba-org/mamba/blob/31b23473e059b47c03b065e5592ea5a7cf77fc9b/libmamba/src/core/transaction_context.cpp#L319

File: dist/setup/index.js
Comment (jaimergp) [inline dist/setup/index.js:1]: Ignore this diff, it's autogenerated.

File: dist/delete/index.js
Comment (jaimergp) [inline dist/delete/index.js:1]: Ignore this diff, it's autogenerated.

**Comment by jaimergp** [comment]
@conda-incubator/setup-miniconda, ready for review!

**Review by dbast — APPROVED** [review]

**Pull request merged**